enhancements/aws-permissions-check-bypass.md

```diff
@@ -0,0 +1,269 @@
+the credentials provided are sufficient for installation and for
+cloud-credential-operator.
+
+The installer will then take the value of this variable to build a ConfigMap
+manifest for the CCO so that it too knows what credentials mode it should
+operator under.
+
+### cloud-credential-operator
+Extend the existing ConfigMap that CCO uses, to allow indicating that the
+permissions checking should be bypassed, and to indicate whether `mint` or
+`passthrough` mode should be assumed.
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cloud-credential-operator-config
+  namespace: openshift-cloud-credential-operator
+  annotations:
+    release.openshift.io/create-only: "true"
+data:
+  force-cloud-credentials-mode: "mint|passthrough"
+```
+
+### User Stories [optional]
+
+#### Story 1
+
+A user installing OpenShift in an AWS account subject to SCPs would run the
+installer with the environment variable set. This will generate manifests for
+CCO containing the ConfigMap definition like the one defined above:
+
+```
+OPENSHIFT_INSTALL_ASSUME_PERMISSIONS_MODE="mint" ./openshift-install create manifests --dir my-aws-cluster
+```
+
+This will cause the installer to skip any pre-flight permissions checks and lay
+down the manifest for CCO to indicate that `mint` mode should be assumed:
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cloud-credential-operator-config
+  namespace: openshift-cloud-credential-operator
+  annotations:
+    release.openshift.io/create-only: "true"
+data:
+  force-cloud-credentials-mode: "mint"
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+Bypassing these checks means that errors will be encountered at the moment the
+API calls are attempted. For example a user with enough permissions to create
+VPCs, Security Groups, and Route53 entries may error when setting up an Elastic
+IP. Now depending on the permissions on the credentials, they may not be
+sufficient to clean up what was created by the installer up until the failure.
+
+Another error case is that the installer is able to complete its portion of the
+bootsrapping, but the CCO and in-cluster AWS API users may fail to come up due
+to lack of sufficient permissions provided to the CCO. These will show up as
+operators unable to reach their installed=true/progressing=false state.
+
+### Risks and Mitigations
+
+Giving a user a way to avoid the dynamic permissions checking means that users
+will need a reliable way to know which permissions are necessary to complete a
+successfull installation (both for `mint` and `passthrough` modes). At present,
+the list of permissions required for an installation are stored as a static list
+of permissions in the installer code, and the permissions needed for `mint` and
+`passthrough` mode are stored in the cloud-credential-operator repo. Publishing
+and updating these lists should become part of the documentation effort for
+OpenShift releases.
```

Comment:
Enhancement I'm working on for disabled CCO may be able to offer some tooling here, I'm hoping the CLI can be enhanced to display and diff the cred requests in a release image.

Reply:
There's also the oc adm release command which could be helpful for the docs on this feature, as well as disabled CCO.